pyanaconda/payload/dnf/payload.py

```diff
@@ -1932,24 +1934,43 @@ def _load_treeinfo_repositories(self, base_repo_url):
             for repo_md in self._install_tree_metadata.get_metadata_repos():
                 if repo_md.path not in existing_urls:
                     repo_treeinfo = self._install_tree_metadata.get_treeinfo_for(repo_md.name)
-                    repo_enabled = repo_treeinfo.type in enabled_repositories_from_treeinfo
+
+                    # disable repositories disabled by user manually before
+                    if repo_md.name in disable_repositories:
+                        repo_enabled = False
+                    else:
+                        repo_enabled = repo_treeinfo.type in enabled_repositories_from_treeinfo
+
                     repo = RepoData(name=repo_md.name, baseurl=repo_md.path,
                                     install=False, enabled=repo_enabled)
                     repo.treeinfo_origin = True
-                    log.debug("Adding new treeinfo repository %s", repo_md.name)
+                    log.debug("Adding new treeinfo repository: %s enabled: %s",
+                              repo_md.name, repo_enabled)
                     self.add_repo(repo)
 
     def _cleanup_old_treeinfo_repositories(self):
         """Remove all old treeinfo repositories before loading new ones.
 
         Find all repositories added from treeinfo file and remove them. After this step new
         repositories will be loaded from the new link.
+
+        :return: list of repositories which were disabled before removal
+        :rtype: [str]
         """
+        disabled_repo_list = []
```

Comment:
It is not clear that it is a list of repo names.

Reply:
I've tried better name.

Reply:
I think it is the same. Could you at least fix the docstring, please?

Reply:
Ohh, sorry I missed that one.